core/src/main/java/io/parsingdata/metal/expression/value/Join.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2013-2023 Netherlands Forensic Institute
+ * Copyright 2021-2023 Infix Technologies B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.expression.value;
+
+import static io.parsingdata.metal.Util.checkContainsNoNulls;
+
+import java.util.Arrays;
+
+import io.parsingdata.metal.data.ImmutableList;
+import io.parsingdata.metal.data.ParseState;
+import io.parsingdata.metal.encoding.Encoding;
+
+/**
+ * A {@link ValueExpression} that joins multiple {@link ValueExpression}s by concatenating
+ * the individual results to a single list.
+ * <p>
+ * A Join expression can have zero or more expressions. If none is provided, this will return an empty list.
+ * Else, each expression is evaluated and concatenated to a single list.
+ */
+public class Join implements ValueExpression {
+
+    private final ValueExpression[] expressions;
+
+    public Join(final ValueExpression... expressions) {
+        this.expressions = checkContainsNoNulls(expressions, "expression");
+    }
+
+    @Override
+    public ImmutableList<Value> eval(final ParseState parseState, final Encoding encoding) {
+        return Arrays.stream(expressions)
+            .map(e -> e.eval(parseState, encoding))
+            .peek(System.out::println)
```

Comment:
Whoops? ;-)

Reply:
Yes, lol! Forgot to commit the change during demo. ;) Removed it!